### ISSUE TITLE
SubViewportContainer processing Key Input Events before other Control-Nodes

### ISSUE THREAD

**Issue by quiyip** [godotengine/godot#81184]
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Windows 10

### Issue description

[The docs](https://docs.godotengine.org/en/stable/tutorials/inputs/inputevent.html) state that inputs are processed in the order of `_input()` > `_gui_input()` > `_shortcut_input()` > `_unhandled_input()` > `_unhandled_key_input()`. However, `SubViewportContainer` overrides this ordering by propagating all input (key) events at `input()`. This triggers all input event functions of its child nodes before all other Control nodes, completely messing up the ordering.

This is a continuation of #58334
although this merge fixed the problem when events contain a `position`, it does not affect key-press events.

(other related issues are #48401 #48368 and #39666)
@Sauermann 

### Steps to reproduce

1) Download reproduction project
2) Run the project
3) Left-click on the `LineEdit` to apply focus
4) Press any button on the keyboard
5) Watch the wrong order appear in the console

Correct ordering is:
1		2			3				4					5
_input() > _gui_input() > _shortcut_input() > _unhandled_input() > _unhandled_key_input()

But instead, the ordering is 1, 3, 4, 5, 2

SceneTree
![image](https://github.com/godotengine/godot/assets/95621115/193ebbd4-23da-45bd-ab5a-b947a5aa94f4)

`LineEdit.gd`
```
extends LineEdit


func _input(event: InputEvent) -> void:
	if event is InputEventKey:
		printt(1, "input")


func _gui_input(event: InputEvent) -> void:
	if event is InputEventKey:
		printt(2, "gui_input")
```

`Player.gd`
```
extends Node2D



func _shortcut_input(event: InputEvent) -> void:
	if event is InputEventKey:
		printt(3, "_shortcut_input")


func _unhandled_input(event: InputEvent) -> void:
	if event is InputEventKey:
		printt(4, "_unhandled_input")


func _unhandled_key_input(event: InputEvent) -> void:
	printt(5, "_unhandled_key_input")
```

### Minimal reproduction project

[SubViewportContainer key input bug reproduction project.zip](https://github.com/godotengine/godot/files/12480486/SubViewportContainer.key.input.bug.reproduction.project.zip)

**Comment by Sauermann** [comment]
I believe, that the problem you describe is a duplicate of #79235 and got fixed in current master by #79248.

Can you please verify, if your problem is still apparent in [Godot V4.2 dev3](https://godotengine.org/article/dev-snapshot-godot-4-2-dev-3/)?

Also please be aware that this order `_input()` > `_gui_input()` > `_shortcut_input()` > `_unhandled_input()` > `_unhandled_key_input()`  is guaranteed only within a single Viewport.

In your description, you are utilizing different Viewports:
- root viewport with `LineEdit`
- SubViewport with `Node2D`

So the above order is not guaranteed for comparing `LineEdit` and `Node2D`.

**Comment by quiyip** [comment]
Thank you, fixed in v4.2.dev3.official [013e8e3af]
